### PR TITLE
Add DocumentInverterCollection.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@baldersheim , @vekterli : FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace search::memoryindex {
+
+class DocumentInverter;
+class DocumentInverterContext;
+
+/*
+ * Class containing the document inverters used by a memory index.
+ */
+class DocumentInverterCollection {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.h:17]: Consider adding a separate unit test for this class to test the management of free, inflight and active inverters.

File: searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace search::memoryindex {
+
+DocumentInverterCollection::DocumentInverterCollection(DocumentInverterContext& context)
+    : _context(context),
+      _free_inverters(),
+      _inflight_inverters(),
+      _active_inverter(std::make_unique<DocumentInverter>(_context)),
+      _num_inverters(1),
+      _max_inverters(4)
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.cpp:15]: Consider adding this as a constructor argument.

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.h (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.h:17]: Unit test added.

File: searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.cpp (hunk 2 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/memoryindex/document_inverter_collection.cpp]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**